### PR TITLE
Aggregate

### PR DESCRIPTION
**Pull request by charlesgobina**
## For this Milestone, I
* Altered the table to add a new species column
* Added the new animal data
* Performed all querries required

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -10,4 +10,22 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+VALUES ('Charmander', 'February 8, 2020', 0, false, 11 );
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES ('Plantmon', 'November 15, 2022', 2, true, 5.7 );
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES ('Squirtle', 'April 2, 1993', 3, false, 12.13 );
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES ('Angemon', 'June 12, 2005', 1, true, 45 );
```
Comment (ZeenatLawal) [inline data.sql]: - [ ] All the weight entries for these animals are negative values, kindly correct it.

File: queries.sql [hunk 2]
```diff
@@ -38,4 +38,88 @@ WHERE name NOT LIKE 'Gabumon';
...
+WHERE weight_kg >= 10.4 AND weight_kg <= 17.3; 
+
+-- Inside a transaction update the animals table by setting the 
+-- species column to unspecified. Verify that change was made. 
+-- Then roll back the change and verify that species columns
+--  went back to the state before transaction.
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+ROLLBACK;
```
Comment (ZeenatLawal) [inline queries.sql]: - [ ] Kindly verify that the change was made before and after the rollback is made as stated in the requirements. e.g
```sql
BEGIN; -- start transaction

UPDATE animals
SET species = 'unspecified'; -- make change

SELECT species from animals; -- verify that change was made
ROLLBACK;

SELECT species from animals; -- verify that change was undone
```

File: queries.sql [hunk 3]
```diff
@@ -38,4 +38,88 @@ WHERE name NOT LIKE 'Gabumon';
...
+SET species = 'digmon'
+WHERE name LIKE '%mon';
+
+-- Update the animals table by setting the species column to
+-- pokemon for all animals that don't have species already set.
+BEGIN;
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL
+COMMIT;
```
Comment (ZeenatLawal) [inline queries.sql]: - [ ] Kindly verify that the change was made before and after the commit is made as stated in the requirements. e.g
```sql
BEGIN;
UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

SELECT species from animals; -- verify that change was made
COMMIT;
SELECT species from animals; -- verify that change persists after commit
```

File: queries.sql [hunk 4]
```diff
@@ -38,4 +38,88 @@ WHERE name NOT LIKE 'Gabumon';
...
+
+-- Now, take a deep breath and... Inside a transaction delete
+-- all records in the animals table, then roll back the transaction.
+BEGIN;
+DELETE FROM animals
+ROLLBACK;
```
Comment (ZeenatLawal) [inline queries.sql]: - [ ] Kindly verify that the change was made before and after the rollback is made as stated in the requirements.

**Review by ZeenatLawal — CHANGES_REQUESTED** [review]
Hi @charlesgobina, 

Good job so far!:clap: :clap:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎊 🎊 
- Gitflow is followed. ✔️
- PR has a title and summary. ✔️


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by charlesgobina** [comment]
Hello @ZeenatLawal,
Thank you for the feedback. I have a little worry. When verifying whether changes were made, is adding the comments as you have done just the only thing to do?

**Comment by ZeenatLawal** [comment]
Hello @charlesgobina, yes that's how you verify. This is done to check whether what you updated or deleted was affected. e.g After deleting all animals you can try fetching all animals to confirm if they were all deleted and after rollback, you can try fetching all animals to confirm if they are all restored.

**Comment by charlesgobina** [comment]
@ZeenatLawal, thank you for the clarification.

File: data.sql [hunk 5]
```diff
@@ -10,4 +10,22 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES ('Squirtle', 'April 2, 1993', 3, false, -12.13 );
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES ('Angemon', 'June 12, 2005', 1, true, -45 );
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) 
+VALUES ('Boarmon', 'June 7, 2005', 7, true, -20.4 );
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
```
Comment (SajjadAhmad14) [inline data.sql]: Please change the values of `weight_kg` to positive as it's asked to change by the previous reviewer. Thanks!

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Charles,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by charlesgobina** [comment]
Hi @SajjadAhmad14, thank you for the feedback. 
There seem to be a misunderstanding. Initially, my weight values were positive values but @ZeenatLawal told me to change them to negative values. [here is her reply](https://github.com/charlesgobina/vetinary-clinic/pull/2#discussion_r814006877). 
In your feedback, you are asking me to change the values back to positive. I have just one chance left. Could you please provide a clear and final response on whether the values are positive or negative?

Best,

**Comment by SajjadAhmad14** [comment]
Hi Charles,
I am sorry I made a mistake. I should have approved your project. If you have a review left you can ask for another review otherwise please let me know I will share this issue with the team!
🙏🙏🙏

**Comment by ZeenatLawal** [comment]
Hi @charlesgobina, to clarify what I requested for the change was to change the weight value to negative for the animals I highlighted which are the first 4 animals. I hope this clarifies it, apologies for the misunderstanding.

**Comment by charlesgobina** [comment]
Hi @SajjadAhmad14,
The fault was mine. I actually did not realize some of the weights were negative and others positive. Sorry for the inconvenience 🙏🏾 . @ZeenatLawal I'm also sorry I misinterpreted your feedback. It's all my fault, I didn't read the requirements carefully.  I'm really sorry for dragging you guys into all this confusion.

**Review by julie-ify — APPROVED** [review]
Hi @charlesgobina, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**